src/common/WorkQueue.cc

```diff
@@ -302,36 +299,34 @@ void ShardedThreadPool::shardedthreadpool_worker(uint32_t thread_index)
 
   while (!stop_threads) {
     if (pause_threads) {
-      shardedpool_lock.lock();
+      std::unique_lock ul(shardedpool_lock);
       ++num_paused;
-      wait_cond.Signal();
+      wait_cond.notify_all();
       while (pause_threads) {
        cct->get_heartbeat_map()->reset_timeout(
 	        hb,
 	        wq->timeout_interval, wq->suicide_interval);
-       shardedpool_cond.WaitInterval(shardedpool_lock,
-	   utime_t(
-	   cct->_conf->threadpool_empty_queue_max_wait, 0));
+       shardedpool_cond.wait_for(
+	 ul,
+	 ceph::make_timespan(cct->_conf->threadpool_empty_queue_max_wait));
```

Comment:
make_timespan(...) involves conversion to/from double. i think std::chrono::seconds(...) would be clearer about the units